universal-application-tool-0.0.1/app/views/applicant/ApplicantProgramInfoView.java

```diff
@@ -0,0 +1,153 @@
+import static j2html.TagCreator.a;
+import static j2html.TagCreator.div;
+import static j2html.TagCreator.h2;
+import static j2html.TagCreator.li;
+import static j2html.TagCreator.span;
+import static j2html.TagCreator.ul;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.inject.Inject;
+import controllers.routes;
+import j2html.tags.ContainerTag;
+import j2html.tags.DomContent;
+import java.util.ArrayList;
+import java.util.Locale;
+import play.i18n.Messages;
+import play.mvc.Http;
+import play.twirl.api.Content;
+import services.MessageKey;
+import services.program.ProgramDefinition;
+import views.BaseHtmlView;
+import views.HtmlBundle;
+import views.components.Accordion;
+import views.style.ApplicantStyles;
+import views.style.BaseStyles;
+import views.style.ReferenceClasses;
+import views.style.Styles;
+
+/** Shows information for a specific program with a button to start the application. */
+public class ApplicantProgramInfoView extends BaseHtmlView {
+
+  private final ApplicantLayout layout;
+
+  @Inject
+  public ApplicantProgramInfoView(ApplicantLayout layout) {
+    this.layout = checkNotNull(layout);
+  }
+
+  public Content render(
+      Messages messages,
+      ProgramDefinition program,
+      Http.Request request,
+      long applicantId,
+      String userName) {
+
+    Locale preferredLocale = messages.lang().toLocale();
+    String programTitle = program.localizedName().getOrDefault(preferredLocale);
+    String programInfo = program.localizedDescription().getOrDefault(preferredLocale);
+
+    HtmlBundle bundle =
+        layout
+            .getBundle()
+            .addMainStyles(Styles.MX_12, Styles.MY_8)
+            .addMainContent(topContent(programTitle, programInfo, messages))
+            .addMainContent(createButtons(applicantId, program.id(), messages));
+
+    return layout.renderWithNav(request, userName, messages, bundle);
+  }
+
+  private ContainerTag topContent(String programTitle, String programInfo, Messages messages) {
+    String programsLinkText = messages.at(MessageKey.TITLE_PROGRAMS.getKeyName());
+    String homeLink = routes.HomeController.index().url();
+    ContainerTag allProgramsDiv =
+        a("<")
+            .withHref(homeLink)
+            .withClasses(Styles.TEXT_GRAY_500, Styles.TEXT_LEFT)
+            .with(span().withText(programsLinkText).withClasses(Styles.PX_4));
+
+    ContainerTag titleDiv =
+        h2().withText(programTitle)
+            .withClasses(
+                BaseStyles.TEXT_SEATTLE_BLUE,
+                Styles.TEXT_2XL,
+                Styles.FONT_SEMIBOLD,
+                Styles.TEXT_GRAY_700,
+                Styles.MT_4);
+
+    // "Markdown" the program description.
+    String[] lines = Iterables.toArray(Splitter.on("\n").split(programInfo), String.class);
+    ImmutableList<DomContent> items = formatText(lines, false);
```

Comment:
(blocking) `formatText` and the other markdown stuff should be pulled out into its own public class that we can add tests to.

Reply:
We should also document what format is supported.

Reply:
Done. Moved it out into another class `TextFormatter` along with `createLinksAndEscapeText`